src/tpm2/crypto/openssl/CryptRsa.c

```diff
@@ -1444,7 +1443,7 @@ CryptRsaDecrypt(
                     ERROR_RETURN(TPM_RC_FAILURE);
                 memcpy(tmp, label->buffer, label->size);
 
-                if (EVP_PKEY_CTX_set0_rsa_oaep_label(ctx, tmp, label->size) <= 0)
+                if (label->size > 0 && EVP_PKEY_CTX_set0_rsa_oaep_label(ctx, tmp, label->size) <= 0)
```

Comment:
I don't think in this case it is necessary to again check label->size > 0. The whole block is already protected with this check.